pkg/phlaredb/phlaredb.go

```diff
@@ -524,3 +525,46 @@ func (f *PhlareDB) BlockMetadata(ctx context.Context, req *connect.Request[inges
 
 	return connect.NewResponse(&result), nil
 }
+
+func (f *PhlareDB) GetProfileStats(ctx context.Context, req *connect.Request[typesv1.GetProfileStatsRequest]) (*connect.Response[typesv1.GetProfileStatsResponse], error) {
+	sp, _ := opentracing.StartSpanFromContext(ctx, "PhlareDB GetProfileStats")
+	defer sp.Finish()
+
+	metas := make([]*block.Meta, 0)
+
+	f.headLock.RLock()
+	for _, h := range f.heads {
+		metas = append(metas, h.meta)
+	}
+	for _, h := range f.flushing {
+		metas = append(metas, h.meta)
+	}
+	f.headLock.RUnlock()
```

Comment:
I'm not 100% sure but access to the `h.meta.MaxTime` and `MinTime` should be only done with the mutex acquired, so there will probably be a race when you latter access meta in `getProfileStatsFromMetas`. It's better to call `h.Bounds()` here

Reply:
Good point, I copied this from `BlockMetadata` but didn't realize that one is using `Bounds()`.